File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.py
class Solution {
    public int numberOfBeams(String[] bank) {
        int ans = 0, prev = 0;
        for(String s : bank){
            int cnt = 0;
            for(Character c : s.toCharArray()){
                if(c == '1') cnt++;
            }
            if(cnt != 0){
                ans += cnt*prev;
                prev = cnt;
            }
        }
        return ans;
    }
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int ans = 0, cnt = 0;
        for(int i=0; i<bank.size(); i++){
            int cnt1 = 0;
            for(int j=0; j<bank[i].size(); j++){
                if(bank[i][j] == '1') cnt1++;
            }
            if(cnt1 != 0){
                ans += cnt * cnt1;
                cnt = cnt1;
            }
        }
        return ans;
    }
};